clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/KubernetesVersionedArtifactConverter.java

```diff
@@ -51,4 +61,38 @@ public KubernetesCoordinates toCoordinates(Artifact artifact) {
   public String getDeployedName(Artifact artifact) {
     return artifact.getName() + "-" + artifact.getVersion();
   }
+
+  private String getVersion(String type, String name, String location) {
+    List<Artifact> priorVersions = artifactProvider.getArtifacts(type, name, location);
+
+    List<Integer> taken = priorVersions.stream()
+        .map(Artifact::getVersion)
+        .filter(Objects::nonNull)
+        .filter(v -> v.startsWith("v"))
+        .map(v -> v.substring(1))
+        .map(v -> {
+          try {
+            return Integer.valueOf(v);
+          } catch (NumberFormatException e) {
+            return null;
+          }
+        } )
+        .filter(Objects::nonNull)
+        .filter(i -> i >= 0)
+        .collect(Collectors.toList());
+
+    taken.sort(Integer::compareTo);
+
+    int attempt = 0;
+    for (Integer exists : taken) {
```

Comment:
Could be a `forEach`.

Reply:
I don't really see the advantage, unless we could just plug a little stateless lambda in.

Reply:
Just Java 8 style is all. Fine as it is.

Reply:
https://stackoverflow.com/a/20177092/1435439

Reply:
Rescinded.